### ISSUE TITLE
isBefore / isAfter / isEqual: WHAT am I doing wrong here!?

### ISSUE THREAD

**Issue by NerdyDeedsLLC** [date-fns/date-fns#2500]
I'm seriously at a bloody loss here.

So I take two dates: `sD` (a stored `startDate` in my class), and `dt` (a dynamically-generated one). For this example, I'll pick dates a couple days apart, so we know there's not timezone tomfoolery taking place. Next, just to be EXTRA sure there's no cross-pollution taking place, I'll rebuild them into fresh, shiny `Date`s using the `new` syntax and their constructors.
```
dt = new Date(dt);              // Result: Mon May 24 2021 00:00:00 GMT-0500 (Central Daylight Time)
sD = new Date(this.startDate);  // Result: Wed May 26 2021 00:00:00 GMT-0500 (Central Daylight Time)
console.log(dt);                // Result: Mon May 24 2021 00:00:00 GMT-0500 (Central Daylight Time)
console.log(sD);                // Result: Wed May 26 2021 00:00:00 GMT-0500 (Central Daylight Time)
```

Capital. So I've generated two new `Date` objects with the constructor, they're manifestly JS `Date`s, and I've confirmed their values. So let's compare:

```
isEqual(sD, dt);    // Expected: false. Result: false
```
Excellent. We've established they're not the same day. Moving along:

```
isBefore(dt, sD);   // Expected: true.  Result: false
isBefore(sD, dt)    // Expected: false. Result: false

isAfter(dt, sD)     // Expected: false. Result: false
isAfter(sD, dt)     // Expected: true.  Result: false
```
Bwuh...?
Now, it's been 25 years since I went to school for this stuff, so maybe my boolean logic is a bit off, but if two dates are not, in fact, the same day, then this suggests one occurs after the other, humanity's linear progression through time being what it is. So _**how on earth am I getting results indicating that both dates occur**_ **neither before nor after** _**the other**_? 

Oh, and just for giggles:
```
isWeekend(dt); // Expected: true.  Result: true          (Note: May 23rd WAS a Sunday...
isWeekend(sD); // Expected: false. Result: false.            ...and May 26th a Wednesday)
```
So they're clearly parseable `Date`s, and being identified correctly. So finally, just to make absolutely SURE I'm totally losing my mind here:
```
formatRFC3339(dt);          // Expected: 2021-05-23T00:00:00-05:00;  Result: 2021-05-23T00:00:00-05:00
formatRFC3339(sD);          // Expected: 2021-05-26T00:00:00-05:00;  Result: 2021-05-26T00:00:00-05:00
let t_dt=dt.getTime();      // Expected: Some big-ass number.        Result: 1621746000000 (Yup. As expected.)
let t_sD=sD.getTime();      // Expected: A DIFFERENT big-ass number. Result: 1622005200000 (Big-ass? ✓ !=? ✓)
console.log(t_dt < t_sD);   // Expected: true.  Result: true
console.log(t_dt === t_sD); // Expected: false. Result: false
console.log(t_dt > t_sD);   // Expected: false. Result: false
```
"Oh, thank goodness; that's a relief⸮"

Now, albeit, this is my first time using this library. But I've managed to successfully execute every other task utilizing it without a problem, and I've been a FED/JS developer since the 90's, so I'm fairly certain I'm at least _running_ the code correctly. Please, for the love of all that's good: _WHAT am I missing here!?_ I'm sure it's SOMETHING galactically-stupid, but... um... 

...please advise?  ¯\\ _ (ツ) _ /¯

**Comment by tan75** [comment]
hi @NerdyDeedsLLC 

which version of date-fns are you using in this example?

**Comment by NerdyDeedsLLC** [comment]
`"date-fns": "^2.22.1",`, per my `package.json`.

I didn't specify when I did the `npm install`; I only learned of the library yesterday, while attempting to do anything in my power to steers a JS away from moment.js

**Comment by tan75** [comment]
I compared those dates and here is the result: 

```
const dt = new Date('Mon May 24 2021 00:00:00 GMT-0500');
const sD = new Date('Wed May 26 2021 00:00:00 GMT-0500');
const result = dateFns.isBefore(dt, sD); //=> true
```

As well, can you please do `console.log(isDate(dt));` to make sure `dt` is an instance of a Date 

Is it possible to put your code out into sandbox so we could see the context of `this`?

**Comment by NerdyDeedsLLC** [comment]
I can tell you off the top of my head `this` will be the class's scope, but I'll double check...

```
isDate(dt) : true
this:  ReportGrid {isInitialized: true, reportName: "Test", startDate: Wed May 26 2021 00:00:00 GMT-0500 (Central Daylight Time), endDate: Wed Jun 09 2021 00:00:00 GMT-0500 (Central Daylight Time), startDayOfWeek: 3, …}
      dataFilePulls: []
      daySpan: 14
      endDate: Wed Jun 09 2021 00:00:00 GMT-0500 (Central Daylight Time) {}
      endDayOfWeek: 3
      isInitialized: true
      nameOfSprint: ""
      reportName: "Test"
      reportSettings: {showWeekends: false, showNonWorkDays: true, excludedDays: Array(0), showGrid: true, showGraph: true}
      startDate: Wed May 26 2021 00:00:00 GMT-0500 (Central Daylight Time) {}
      startDayOfWeek: 3
      weekSpan: 2
      workdaySpan: 10

```

**Comment by NerdyDeedsLLC** [comment]
sandbox will be trickier... it's one of several dozen files that all need to be present to get to the execution point, and truth to tell, I'm not actually sure my NDA would allow it. I'm happy to post the whole class as it stands right now, if you think that'll help.

**Comment by NerdyDeedsLLC** [comment]
Actually, I figured it out... but this is a bizarre thing:

So I was reassembling a MVP version of the code that I COULD post out to you. I was trying to reduce the extraneous bits as far as possible so as not to confuse the issue, re-testing after each removal. The problem persisted until I changed
```
import parseISO from 'date-fns/parseISO'
import isEqual from 'date-fns/isEqual'
import isBefore from 'date-fns/isBefore'
import isAfter from 'date-fns/isAfter'
import subDays from 'date-fns/subDays'
```
...to...
```
import {subDays, parseISO, isEqual, isBefore, isAfter} from 'date-fns'
```
...whereupon, lo and behold...

Consequently, I'm now inclined to think this is a problem with the package-bundler, and not you guys at all. The team working on this bit that had asked me for an assist _is_ using an older version of Rollup vinyl'd through Gulp somehow (I haven't investigated yet). Best guess at this point was it was reinitializing the package? Supplanting one method with another? I'm not wholly sure. I do know that if I reduce the import to a single line it works as expected. Moreover, after seeing this, I initialized a blank Webpack project and therein the multi-line works as well. 

Sorry for the false alarm; not your problem by the looks of things! Thanks for the absolutely amazing response turnaround @tan75 . I can see why this library came so highly recommended with volunteers like you.
